### PR TITLE
fix: support skipping tags

### PR DESCRIPTION
**Pull request by nodece**
## Motivation

I'm using the buildx to build the multiple-arch images, when I use `-Ddocker.skip.tag=true` to skip the tag, this plugin still tags the image.

## Changes

- Add skip tag to the image level by `<skipTag>`
- When the skip tag is `true`, the plugin will not tag the image on the build and buildx stage

## Doc

The previous document included `<skipTag>`, so I did not add the document.

### CODE REVIEW

File: src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java [hunk 1]
```diff
@@ -190,6 +190,9 @@ public class BuildImageConfiguration implements Serializable {
     @Parameter
     private Boolean skipPush;
 
+    @Parameter
+    private Boolean skipTag;
```
Comment (rohanKanojia) [inline src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java:194]: Umm, is this new field really needed here? Do you want to skip tag at image level?

Comment (nodece) [reply src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java:194]: > Do you want to skip tag at image level?

When using the buildx to build the multi-arch with push action, I do not want to push tag.

Right now, this PR supports skip tag at image level.

File: src/main/java/io/fabric8/maven/docker/service/BuildXService.java [hunk 2]
```diff
@@ -142,10 +142,12 @@ protected void buildX(List<String> buildX, String builderName, BuildDirs buildDi
...
-        buildConfiguration.getTags().forEach(t -> {
-            cmdLine.add("--tag");
-            cmdLine.add(new ImageName(imageConfig.getName(), t).getFullName(configuredRegistry));
-        });
+        if (!buildConfiguration.skipTag()) {
+            buildConfiguration.getTags().forEach(t -> {
+                cmdLine.add("--tag");
+                cmdLine.add(new ImageName(imageConfig.getName(), t).getFullName(configuredRegistry));
+            });
+        }
```
Comment (rohanKanojia) [inline src/main/java/io/fabric8/maven/docker/service/BuildXService.java:150]: Is it possible to add a test case for scenario where `buildConfiguration.skipTag()` evaluates to `true`?

File: src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java (hunk 1 above)
Comment (rohanKanojia) [reply src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java:194]: okay, I see your point now. `skipTag` configuration option is not available in BuildXService

File: src/main/java/io/fabric8/maven/docker/service/BuildXService.java (hunk 2 above)
Comment (nodece) [reply src/main/java/io/fabric8/maven/docker/service/BuildXService.java:150]: I will add this test case tomorrow.

**Comment by nodece** [comment]
@rohanKanojia Thank you for your review, this PR has been updated, could you review this PR again?

**Comment by nodece** [comment]
MacOS ci failed, what do you suggest, it seems to be unable to pull pg image.

File: src/main/java/io/fabric8/maven/docker/BuildMojo.java [hunk 3]
```diff
@@ -105,7 +105,9 @@ private void proceedWithDockerBuild(ServiceHub hub, BuildService.BuildContext bu
             } else {
                 buildService.buildImage(imageConfig, pullManager, buildContext, buildArchiveFile);
                 if (!skipTag) {
-                    buildService.tagImage(imageConfig);
+                    if (!imageConfig.getBuildConfiguration().skipTag()) {
```
Comment (rohanKanojia) [inline src/main/java/io/fabric8/maven/docker/BuildMojo.java:108]: Can this `if` statement merged with the one above?

Comment (nodece) [reply src/main/java/io/fabric8/maven/docker/BuildMojo.java:108]:
```
if (!skipTag && !imageConfig.getBuildConfiguration().skipTag()) {
   buildService.tagImage(imageConfig);
}
```

It looks like I should use `&&`, is it correct?

Comment (rohanKanojia) [reply src/main/java/io/fabric8/maven/docker/BuildMojo.java]: Yes, looks correct to me.

**Comment by rohanKanojia** [comment]
@nodece : Could you please address[ Sonar Code smells](https://sonarcloud.io/project/issues?id=fabric8io_docker-maven-plugin&pullRequest=1787&resolved=false&sinceLeakPeriod=true) that seem applicable?

**Comment by nodece** [comment]
@rohanKanojia Done.

**Comment by nodece** [comment]
should I upgrade the postgres image?  It looks blocked my PR.

File: src/test/java/io/fabric8/maven/docker/BuildMojoTest.java [hunk 4]
```diff
@@ -241,6 +244,75 @@ void buildUsingBuildxWithMultipleAuth() throws IOException, MojoExecutionExcepti
...
+
+    @Test
+    void buildWithTag() throws IOException, MojoExecutionException {
+        buildWithTag(false);
+    }
+
+    @Test
+    void buildWithSkipTag() throws IOException, MojoExecutionException {
+        buildWithTag(true);
+    }
```
Comment (rohanKanojia) [inline src/test/java/io/fabric8/maven/docker/BuildMojoTest.java:314]: Is it possible to somehow use `@Nested` annotation to group normal build and buildX tests? Initially, I was a bit confused by the test names.  Can we use JUnit5 parameterized tests here to reduce duplication?

Comment (nodece) [reply src/test/java/io/fabric8/maven/docker/BuildMojoTest.java]: @rohanKanojia Done.

**Comment by rohanKanojia** [comment]
@nodece : oh, that's quite strange. I see that this field got added in this PR.

**Comment by rohanKanojia** [comment]
The documentation was wrongly added in https://github.com/fabric8io/docker-maven-plugin/pull/954

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**